### PR TITLE
New "Research directions" section for the main page; ML project page

### PR DESCRIPTION
**Pull request by naummo**
- Added a new section for the main page, where projects are described by two-three sentences and are shown in a 3-column grid.
- Swapped section background colours so that with the introduction of new section the background behind mugshots remains white.
- Added a page about the Machine Learning project with a description and links to media files.

Changes to CSS files affect only the individual project pages and the new section on the main page.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -303,6 +281,45 @@
...
+              <!--
+              NOTE: fill this section next.
+              -->
+              &nbsp;
+          </section>
+          <section class="4u">
+            <span class="pennant"><span class="fa fa-connectdevelop"></span></span>
+              <h3>Machine Learning</h3>
+              <p>Making neural networks (NNs) performance-portable using <span class="lift-bold">Lift</span>: implementing generic and NN-specific optimizations as rewrite rules for efficient hardware utilization;
+        introducing traditional NN building blocks such as <i>conv</i>, <i>norm</i> and <i>fully_connected</i> for seemless integration with popular machine learning libraries.</p>
```
Comment (michel-steuwer) [inline index.html]: for seamless integration of Lift with 
       ^^ typo here             ^^ add "of Lift"

Comment (naummo) [inline index.html]: Done, thanks!

**Comment by cdubach** [comment]
Looks good. Could you add a drop-down menu for the "research directions" in the top bar showing all the sub pages?

**Comment by naummo** [comment]
@cdubach: Sure, will do in a separate pull request.

**Pull request merged**